MR4/expression/call.3.mr

```diff
@@ -0,0 +1,99 @@
+## MR4 compiler - Call expression
+
+## Function call expression
+class Call-expression(Expression)
+  owner Expression function
+  owner Call-arguments arguments
+  
+  func inst parse-new(user String ends):(
+      owner Expression expression, copy Char end)
+    new Call-expression call-expression
+    call-expression.parse(owner expression):(copy end)
+    expression := call-expression
+  
+  func inst parse(owner Expression function):(copy Char end)
+    self.function := function
+    new Call-arguments in self.arguments
+    self.arguments.parse():(copy end)
+  
+  func dynamic write()
+    self.function.write()
+    self.arguments.write()
+
+
+static Call-arguments
+  owner Call-argument parameters
+  owner Call-argument outputs
+  
+  ## parsing `(access value, ...):(access value, ...)`
+  func inst parse():(copy Char end)
+    self.parse-args():(owner self.parameters, copy end)
+    if end = ':'
+      read-expect(user "(")
+      self.parse-args():(owner self.outputs, copy end)
+  
+  ## parsing single argument list `(access value, ...)`
+  func inst parse-args():(owner Call-argument first, copy Char end)
+    user Call-argument arg
+    var String{128} access-str
+    read-until(user access-str, user " )", copy false):(copy end)
+    if end = '\n' and access-str.length = 0
+      read-line-break-spaces()
+      read-until(user access-str, user " ", copy false):(copy end)
+    if end != ')' or access-str.length > 0
+      do
+        if access-str.length = 0
+          f-syntax-error-c(user "expected access, got", copy end)
+        if end != ' '
+          f-syntax-error-c(user "expected space, got", copy end)
+        if ? arg
+          new Call-argument in arg.next
+          arg := arg.next
+        else
+          new Call-argument in first
+          arg := first
+        arg.parse(user access-str):(copy end)
+        while end = ','
```

Comment:
I don't understand the `do..while` syntax, even after reading https://github.com/meircif/mr-dev/wiki/MR-Syntax#control-flow